app/controllers/actions/DataRetrievalAction.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.actions
+
+import controllers.auth.{DataRequest, IdentifierRequest}
+import play.api.mvc.ActionTransformer
+import repository.JourneyCacheNewRepository
+import uk.gov.hmrc.tai.model.UserAnswers
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class DataRetrievalActionImpl @Inject() (
+  val journeyCacheNewRepository: JourneyCacheNewRepository
+)(implicit val executionContext: ExecutionContext)
+    extends DataRetrievalAction {
+
+  override protected def transform[A](request: IdentifierRequest[A]): Future[DataRequest[A]] =
+    journeyCacheNewRepository
+      .get(request.userId)
```

Comment:
The id for the journey cannot be the session only. When helping someone the data will be different within the same session.

Reply:
Fixed